src/algebra/field_power.lean

```diff
@@ -212,8 +212,8 @@ lemma fpow_strict_mono {x : K} (hx : 1 < x) :
   strict_mono (λ n:ℤ, x ^ n) :=
 λ m n h, show x ^ m < x ^ n,
 begin
-  have xpos : 0 < x := by linarith,
-  have h₀ : x ≠ 0 := by linarith,
+  have xpos : 0 < x := zero_lt_one.trans hx, --by linarith, -- worked!
+  have h₀ : x ≠ 0 := by linarith, --xpos.le,
```

Comment:
Is linarith still broken here?

Reply:
No, I just checked: `linarith` works, but I still left the actual proofs, since they appear easier than the `linarith` proof.  If you think that it would be better to leave `by linarith`, I will revert this change as well!

Reply:
It might make sense for you to extract some of your "it's more readable without linarith" and "I tidied up this long proof" bits from this PR to a new PR. From what I can tell, the new proofs don't really rely on the refactor much.

Reply:
Ah, thank you!  Now that you mention this, it is clearly the correct strategy, I just feel still a little awkward with having a lot of PRs open at once, with some dependencies!

Reply:
For reference, PR #7831 implements the changes without linarith and the parentheses in the other file.